container-registry/container-analysis/spec/sample_spec.rb

```diff
@@ -0,0 +1,287 @@
+    occurrence_id = Pathname.new(created.name).basename.to_s
+    delete_occurrence occurrence_id: occurrence_id, project_id: @project_id
+    delete_note project_id: @project_id, note_id: note_id
+  end
+
+  example "test find vulnerabilities" do
+    result_list = find_vulnerabilities_for_image resource_url: @image_url, project_id: @project_id
+    c = result_list.count
+    expect(c).to eq(0)
+
+    # create vulnerability occurrence
+    create_occurrence resource_url: @image_url, 
+                      note_id: @note_id,
+                      occurrence_project: @project_id,
+                      note_project: @project_id
+    try = 0
+    while c != 1 and try < @try_limit
+      sleep @sleep_time
+      result_list = find_vulnerabilities_for_image resource_url: @image_url, project_id: @project_id
+      c = result_list.count
+      try += 1
+    end
+    expect(result_list.count).to eq(1)
+  end
+
+  example "test find high severity vulnerabilities" do
+    result_list = find_high_severity_vulnerabilities_for_image resource_url: @image_url, project_id: @project_id
+    sleep 1
+    expect(result_list.count).to eq(0)
+ 
+    # create vulnerability occurrence
+    create_occurrence resource_url: @image_url, 
+                      note_id: @note_id,
+                      occurrence_project: @project_id,
+                      note_project: @project_id
+    result_list = find_high_severity_vulnerabilities_for_image resource_url: @image_url, project_id: @project_id
+    sleep 1
+    expect(result_list.count).to eq(0)
+
+    # create critical secerity occurrence
+    note_id = "severe-note-" + @uuid
+    formatted_project = Grafeas::V1::GrafeasClient.project_path(@project_id)
+    note = {vulnerability: {severity: :CRITICAL}}
+    note = { 
+      vulnerability: {
+        severity: :CRITICAL,
+        details: [
+            affected_cpe_uri: 'your-uri-here',
+            affected_package: 'your-package-here',
+            min_affected_version: { kind: Grafeas::V1::Version::VersionKind::MINIMUM },
+            fixed_version: { kind: Grafeas::V1::Version::VersionKind::MAXIMUM }
+          ],
+        } 
+      }
+    client.create_note(formatted_project, note_id, note)
+    formatted_note = Grafeas::V1::GrafeasClient.note_path(@project_id, note_id)
+    occurrence = {note_name: formatted_note, 
+                  vulnerability: {severity: :CRITICAL},
+                  resource:{uri: @image_url}}
+    occurrence = { 
+      note_name:     formatted_note,
+      resource_uri: @image_url,
+      vulnerability: {
+        severity: :CRITICAL,
+        package_issue: [
+          affected_cpe_uri: 'your-uri-here',
+          affected_package: 'your-package-here',
+          min_affected_version: { kind: Grafeas::V1::Version::VersionKind::MINIMUM },
+          fixed_version: { kind: Grafeas::V1::Version::VersionKind::MAXIMUM }
+        ]
+      }
+    }
+    client.create_occurrence(formatted_project, occurrence)
+
+    # try again
```

Comment:
This block could use more descriptive comments and variables (`try`, `c`)

For example
```rb
# Retry until we find an image with a high severity vulnerability
retry_count = 0
vulnerability_count = 0
```

Reply:
Still need a better comment.